File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;
int count = 0;

void dfs(string word, string target, string current) {
    if (current == target) {
        answer = count;
    }
    
    if (current.length() >= 5) return;
    
    for(int i = 0; i < word.length(); i++) {
        count++;
        dfs(word, target, current + word[i]);
        if (answer != 0) return;
    }
}

int solution(string word) {
    string aeious = "AEIOU";
    dfs(aeious, word, "");
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    int answer = 0;
    int cnt = 0;
    public int solution(String word) {
        dfs("", word);
        return answer;
    }
    
    private void dfs(String word, String word2) {
        if (word.equals(word2)) answer = cnt;
        if (word.length() == 5) {
            return;
        }
        for (char c : new char[]{'A', 'E', 'I', 'O', 'U'}) {
            cnt++;
            dfs(word + c, word2);
        }
    }
}